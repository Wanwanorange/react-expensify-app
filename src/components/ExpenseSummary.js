import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpenseSummary = ({ expenseCount, expensesTotal}) => {
  const expenseFormatter = expenseCount === 1 ? 'expense' : 'expenses';
  const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00');
  return (
  <div>
    <p>
      Viewing {expenseCount} {expenseFormatter} totalling {formattedExpensesTotal}
    </p>
  </div>
)};

const mapStateToProps = (state) => ({
  expenseCount: selectExpenses(state.expenses, state.filters).length,
  expensesTotal: selectExpensesTotal(selectExpenses(state.expenses, state.filters))
});

export default connect(mapStateToProps)(ExpenseSummary);