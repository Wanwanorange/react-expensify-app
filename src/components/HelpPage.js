import React from 'react';

const HelpPage = () => (
  <div>halp</div>
);

export default HelpPage;