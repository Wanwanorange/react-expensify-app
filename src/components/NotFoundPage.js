import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
  // Client side routing: use Link for internal rerouting
  <div>
    404 :( <Link to="/">Go home!</Link>
  </div>
);

export default NotFoundPage;