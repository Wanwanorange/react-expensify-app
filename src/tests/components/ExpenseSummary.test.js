import React from 'react';
import { ExpenseSummary } from '../../components/ExpenseSummary';
import { shallow } from 'enzyme';
import expenses from '../fixtures/expenses';

test('should correctly display message for one expense', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={1} expensesTotal={235}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly display message for multiple expenses', () => {
  const wrapper = shallow(<ExpenseSummary expenseCount={30} expensesTotal={23098725}/>);
  expect(wrapper).toMatchSnapshot();
});