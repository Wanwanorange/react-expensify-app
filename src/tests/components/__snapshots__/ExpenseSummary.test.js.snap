// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly display message for multiple expenses 1`] = `
<div>
  <p>
    Viewing 
    30
     
    expenses
     totalling 
    $230,987.25
  </p>
</div>
`;

exports[`should correctly display message for one expense 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling 
    $2.35
  </p>
</div>
`;
